cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -49,6 +50,59 @@ Status CheckRelCommon(const RelMessage& rel) {
   return Status::OK();
 }
 
+Result<fs::FileInfoVector> GetGlobFiles(const std::shared_ptr<fs::FileSystem>& filesystem,
+                                        std::string& path) {
```

Comment:
Minor nit: should this parameter be named `glob`?

Reply:
Also, should be `const std::string&`